tests/test_configuration.py

```diff
@@ -29,14 +29,17 @@ def test_returns_default_when_no_value_set(self) -> None:
         assert cfg.value == 42
 
     def test_raises_missing_configuration_when_value_is_none(self) -> None:
-        cfg = EnvValue("test")
+        cfg = EnvValue[str]("test")
 
         with pytest.raises(MissingConfigValue, match="'test' was not defined."):
             cfg.value  # pylint: disable=pointless-statement
 
     @patch.object(EnvValue, "getter", MagicMock(return_value=1))
     def test_modifies_config_value_using_declared_parser_function(self) -> None:
-        cfg = EnvValue("RANDOM_VAR", lambda a: a * 10)
+        def times_10_parser(value: int) -> int:
```

Comment:
`value` should be always a string, as it is impossible to get something else from env:
```suggestion
        def times_10_parser(value: str) -> int:
```